Sources/NIO/SocketAddresses.swift

```diff
@@ -38,6 +38,8 @@ public enum SocketAddressError: Error {
     case unixDomainSocketPathTooLong
     /// Unable to parse a given IP string
     case failedToParseIPString(String)
+    /// Unable to parse a given IP ByteBuffer
+    case failedToParseIPByteBuffer(ByteBuffer)
```

Comment:
I'm afraid we can't add new cases to enumerations: doing so is a Semver major change. Can we add a new `struct` to represent this error instead?

Reply:
I'm having trouble implementing the `struct`. 
Could you enlighten me on this point ?

This is what i did :

```swift
public enum SocketAddressError: Error {
    /// The host is unknown (could not be resolved).
    case unknown(host: String, port: Int)
    /// The requested `SocketAddress` is not supported.
    case unsupported
    /// The requested UDS path is too long.
    case unixDomainSocketPathTooLong
    /// Unable to parse a given IP string
    case failedToParseIPString(String)
    /// Unable to parse a given IP ByteBuffer
    struct failedToParseIPByteBuffer: Error {
      let address: ByteBuffer
    }
}
// ...
throw SocketAddressError.failedToParseIPByteBuffer(address: packedIpAddress)
```

Reply:
```swift
extension SocketAddressError {
    public struct FailedToParseIPByteBuffer: Error, Hashable {
        public var address: ByteBuffer

        public init(address: ByteBuffer) {
            self.address = address
        }
    }
}

throw SocketAddressError.FailedToParseIPByteBuffer(address: packedIpAddress)
```